test/e2e/secretgencontroller/private_registry_auth_test.go

```diff
@@ -0,0 +1,260 @@
+  - name: carvel
+  categories:
+  - testing
+  supportDescription: "Description of support provided for the package"
+---
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: Package
+metadata:
+  name: pkg.test.carvel.dev.1.0.0
+  namespace: %[1]s
+spec:
+  refName: pkg.test.carvel.dev
+  version: 1.0.0
+  template:
+    spec:
+      fetch:
+      - imgpkgBundle:
+          image: registry-svc.%[3]s.svc.cluster.local:443/secret-test/test-bundle
+      template:
+      - ytt: {}
+      - kbld:
+          paths:
+          - "-"
+          - ".imgpkg/images.yml"
+      deploy:
+      - kapp: {}
+---
+apiVersion: packaging.carvel.dev/v1alpha1
+kind: PackageInstall
+metadata:
+  name: %[2]s
+  namespace: %[1]s
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/packageinstalls
+spec:
+  syncPeriod: 30s
+  serviceAccountName: kappctrl-e2e-ns-sa
+  packageRef:
+    refName: pkg.test.carvel.dev
+    versionSelection:
+      constraints: 1.0.0
+`, env.Namespace, name, registryNamespace) + sas.ForNamespaceYAML()
+
+	secretYaml := fmt.Sprintf(`
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: regcred
+type: kubernetes.io/dockerconfigjson
+stringData:
+  .dockerconfigjson: |
+    {
+      "auths": {
+        "registry-svc.%s.svc.cluster.local:443": {
+          "username": "testuser",
+          "password": "testpassword",
+          "auth": ""
+        }
+      }
+    }
+---
+apiVersion: secretgen.k14s.io/v1alpha1
+kind: SecretExport
+metadata:
+  name: regcred
+spec:
+  toNamespaces:
+  - %s
+`, registryNamespace, env.Namespace)
+
+	cleanUp := func() {
+		// Delete App with kubectl first since kapp
+		// deletes ServiceAccount before App
+		kubectl.RunWithOpts([]string{"delete", "apps/" + name}, e2e.RunOpts{AllowError: true})
+		kapp.Run([]string{"delete", "-a", registryName, "-n", registryNamespace})
+		kapp.Run([]string{"delete", "-a", name})
+		kapp.Run([]string{"delete", "-a", "secret-export"})
+	}
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("deploy registry with self signed certs", func() {
+		kapp.Run([]string{"deploy", "-f", "../assets/registry/registry2.yml", "-f", "../assets/registry/certs-for-skip-tls.yml", "-f", "../assets/registry/htpasswd-auth", "-f", "../assets/registry/registry-contents.yml", "-a", registryName})
+	})
+
+	logger.Section("Create Docker Registry Secret", func() {
+		kapp.RunWithOpts([]string{"deploy", "-a", "secret-export", "-f", "-"},
+			e2e.RunOpts{StdinReader: strings.NewReader(secretYaml)})
+	})
+
+	logger.Section("Create PackageInstall", func() {
+		kapp.RunWithOpts([]string{"deploy", "-a", name, "-f", "-"},
+			e2e.RunOpts{StdinReader: strings.NewReader(pkgiYaml)})
+		out := kapp.Run([]string{"inspect", "-a", name, "--raw", "--tty=false", "--filter-kind=PackageInstall"})
+
+		var cr v1alpha12.PackageInstall
+		err := yaml.Unmarshal([]byte(out), &cr)
+		if err != nil {
+			t.Fatalf("Failed to unmarshal: %s", err)
+		}
+
+		require.Equal(t, "", cr.Status.UsefulErrorMessage)
```

Comment:
seems unnecessary given below waits for success and check on configmap.

Reply:
It's mainly to check that the PackageInstall never starts in a failing state. If we only rely on the kubectl wait, then it's possible it will start failing but eventually recover after the placeholder secret is populated.